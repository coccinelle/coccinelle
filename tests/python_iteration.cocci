virtual after_start

@initialize:python@
@@

seen = set()

@r@
identifier fn;
position p;
@@

fn@p(...) { <+... return (ERR(...)); ...+> }

@script:python depends on r@
fn << r.fn;
p << r.p;
@@

if fn not in seen:
    seen.add(fn)
    it = Iteration()
    it.set_files([p[0].file])
    it.add_virtual_rule(after_start)
    it.add_virtual_identifier(err_function, fn)
    it.register()

@e depends on after_start exists@
identifier virtual.err_function;
expression x;
position p;
@@

x@p = err_function(...)

@script:python@
p << e.p;
@@

cocci.print_main("call", p)
