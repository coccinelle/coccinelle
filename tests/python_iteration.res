void f(void) {
    return ERR(42);
}

void g(void) {
    x = f();
}
