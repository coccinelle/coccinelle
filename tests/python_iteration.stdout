* TODO [[view:tests/python_iteration.c::face=ovl-face1::linb=6::colb=4::cole=5][call tests/python_iteration.c::6]]
